### PR TITLE
chores: switch to python-lsp-server and fix typing

### PR DESCRIPTION
**Pull request by ganwell**
My typing rules for this project (there are no company rules):

- In tests `# type: ignore` is allowed if fixing the issue does not help production-code
- In production-code `# type: ignore` is allowed if a manual check happens or
  we can guarantee some runtime behavior. For example we guarantee that a
  weakref is always valid. Similar to `unsafe` in rust.

### CODE REVIEW

File: manabi/lock.py [hunk 1]
```diff
@@ -4,18 +4,33 @@
...
 import traceback
 import weakref
+from abc import ABC, abstractmethod
 from collections.abc import MutableMapping
 from pathlib import Path
+from typing import Callable
 
 import psycopg2.extensions
 from psycopg2 import InterfaceError, OperationalError, connect
+from psycopg2.extensions import connection as PsycopConnection
```
Comment (winged) [inline manabi/lock.py:14]:
```suggestion
from psycopg2.extensions import connection as PsycopgConnection
```

Renamed type should match the package's name

File: manabi/lock.py [hunk 2]
```diff
@@ -4,18 +4,33 @@
...
 from wsgidav.util import get_module_logger  # type: ignore
 
 _logger = get_module_logger(__name__)
 
 
 class ManabiTimeoutMixin:
+    _max_timeout: int = 0
+
+    @property
+    def max_timeout(self) -> float:
```
Comment (winged) [inline manabi/lock.py:25]:
```suggestion
    def max_timeout(self) -> int:
```

Setter wants int, default is int, so this should probably be int too

**Review by winged — CHANGES_REQUESTED** [review]
LGTM, just some minor adjustments required

File: manabi/lock.py (hunk 2 above)
Comment (ganwell) [reply manabi/lock.py:25]: Thanks for spotting that, its the other way round it should be float. I am passing in float somewhere.

**Review by winged — APPROVED** [review]

**Comment by winged** [comment]
Alright, LGTM now. Are you gonna squash the commits before merging?

**Pull request merged**